ironfish/src/rpc/adapters/tcpAdapter.test.ts

```diff
@@ -3,64 +3,64 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 /* eslint-disable jest/no-try-expect */
 /* eslint-disable jest/no-conditional-expect */
-import os from 'os'
 import * as yup from 'yup'
-import { IronfishSdk } from '../../sdk'
-import { RpcRequestError, RpcSocketClient } from '../clients'
-import { ALL_API_NAMESPACES, ApiNamespace } from '../routes'
+import { Assert } from '../../assert'
+import { createNodeTest } from '../../testUtilities'
+import { RpcRequestError } from '../clients'
+import { RpcTcpClient } from '../clients/tcpClient'
+import { ALL_API_NAMESPACES, API_NAMESPACES_PROTECTED } from '../routes'
 import { ERROR_CODES, ValidationError } from './errors'
 import { RpcTcpAdapter } from './tcpAdapter'
 
 describe('TcpAdapter', () => {
-  let tcp: RpcTcpAdapter
-  let sdk: IronfishSdk
-  let client: RpcSocketClient
-
-  beforeEach(async () => {
-    const dataDir = os.tmpdir()
-
-    sdk = await IronfishSdk.init({
-      dataDir: dataDir,
-      configOverrides: {
-        enableRpcTcp: true,
-        enableRpcIpc: false,
-        enableRpcTls: false,
-      },
-    })
-
-    const node = await sdk.node()
+  const nodeTest = createNodeTest(false, {
+    config: {
+      rpcTcpPort: 0,
+      enableRpcTcp: true,
+      enableRpcIpc: false,
+      enableRpcTls: false,
+    },
+  })
 
-    tcp = node.rpc.adapters[0] as RpcTcpAdapter
+  let adapter: RpcTcpAdapter
+  let client: RpcTcpClient
 
-    client = sdk.client
+  beforeEach(() => {
+    Assert.isInstanceOf(nodeTest.node.rpc.adapters[0], RpcTcpAdapter)
+    adapter = nodeTest.node.rpc.adapters[0]
   })
 
   afterEach(async () => {
     client.close()
-    await tcp.stop()
+    await adapter.stop()
   })
 
   it('should send and receive message', async () => {
-    expect(tcp).toBeInstanceOf(RpcTcpAdapter)
-    tcp.router?.register('foo/bar', yup.string(), (request) => {
+    adapter.router?.register('foo/bar', yup.string(), (request) => {
       request.end(request.data)
     })
 
-    await tcp.start()
+    await adapter.start()
+
+    Assert.isNotNull(adapter.addressPort)
+    client = new RpcTcpClient('localhost', adapter.addressPort)
```

Comment:
Its' a little funky that you call this constructor instead of using `sdk.client`, as the latter is used by a lot of code and structured differently. However, I don't see an easy alternative as it may be cludgy to pass the port stuff into `sdk`